assets/onboarding/data/actions.js

```diff
@@ -142,6 +143,7 @@ export function* submitStep( step, stepData, { onSuccess, onError } ) {
 			data: stepData,
 		} );
 		yield successSubmit();
+		yield applyStepData( step, stepData );
```

Comment:
What about we add this side-effect in the controls tied to the `SET_STEP_DATA` instead of creating a new action? Looks like we are duplicating part of the `setStepData` with the name `applyStepData`. Maybe it can look a little confusing what each one does. WDYT?

Reply:
Sounds good, I'll explore

Reply:
Started looking into this and remembered the reason I separated them:  the reducer will not run for the action if there is a control handling it. I also wanted to name it `useStepData`, but that triggered all kinds of ESLint warnings assuming it was a React hook. But I'm open to suggestions on a better name :)

Reply:
Hum! TIL! So for now, it's okay for me ;)